seccomp/umount.go

```diff
@@ -63,7 +64,12 @@ func (u *umountSyscallInfo) process() (*sysResponse, error) {
 		// This restriction should be fine because unmounting /proc inside the
 		// container is not a useful thing to do anyways. Same rationale applies
 		// to "/sys".
-		if u.Target == "/proc" || u.Target == "/sys" {
+		//
+		// Also, notice that we want to clearly differentiate the root /proc and
+		// /sys mounts from those that are present within 'chroot'ed contexts. In
+		// the later case we want to allow users to mount (and umount) both /proc
+		// and /sys file-systems.
+		if (u.Target == "/proc" || u.Target == "/sys") && (u.syscallCtx.root == "/") {
```

Comment:
Should the u.Target == "/sys" check be moved under case "sysfs" below?

Reply:
Why?

The idea is to decide, as early as possible, if sysbox-fs is handling the umount or not -- this is critical-path during container initialization & removal.

Reply:
"u.syscallCtx.root" variable is correctly pointing to "/" to refer to paths to be mount (or mounted) in the host system. See that "root" attribute holds the process' root-path from a host perspective, not from a process one.